### PR TITLE
Vendorize theme

### PR DESCRIPTION
**Pull request by chadwcarlson**
## Why

This PR seems relatively large, but in reality it's mainly re-organizing files around the Hugo theme used to build the docs site.

Namely,

- The `psh-docs` theme is moved top-level, into `themes/psh-docs`
- The `docs.platform.sh` site files are moved from `docs` into `sites/platform`
- The two changes allow us to build a parallel set of docs from `sites/friday` for the fictional company "Deploy Friday". That site is able to reuse 1) content from `sites/platform/src`, 2) data from `shared/data`, and 3) a common theme in `theme/psh-docs`

This is a first variation for allowing vendorization of the public docs site. More experiments and changes to come. 

- Closes https://github.com/platformsh/platformsh-docs/issues/3312
- Closes https://github.com/platformsh/platformsh-docs/issues/3329

Verification:

- [Existing public docs](https://whitelabel-zkk7ody-652soceglkw4u.eu-3.platformsh.site/)
- [Deploy Friday docs](https://docs.deployfriday.net.whitelabel-zkk7ody-652soceglkw4u.eu-3.platformsh.site/)

## What's changed

- [x] theme is cleaned up and further separated
     - (i.e. https://github.com/platformsh/platformsh-docs/issues/3329)
- [x] Public docs acts as primary url
- [x] Basic PoC subsite included that merges main content into custom content
- [x] Search for each, using same service
- [x] Fix internal links test

## Bugs to address

- [x] [on each language page, the example in the Supported versions > Specify the language section that should be pulled in from the language-specification.md  shortcode isn't displayed.](https://docs.deployfriday.net.whitelabel-zkk7ody-652soceglkw4u.eu-3.platformsh.site/languages/elixir.html#specify-the-language)
- [x] [On each service page, the examples from the Example Configuration section aren't displayed correctly (I believe this should be pulled in from the config_links.html  partial via the endpoint-description.md shortcode).](https://docs.deployfriday.net.whitelabel-zkk7ody-652soceglkw4u.eu-3.platformsh.site/add-services/headless-chrome.html#example-configuration)
- [x] [Tier availability callout formatting is unreadable](https://docs.deployfriday.net.whitelabel-zkk7ody-652soceglkw4u.eu-3.platformsh.site/create-apps/source-operations.html)
- [x] The search index cron job has wiped, but not re-indexed the site (Deploy Friday). Fine for main docs site. 
- [x] [Need to update changed files path](https://github.com/platformsh/platformsh-docs/blob/18748590398514ae356c9f285ce40537b75712e3/.github/workflows/get-pr-info.yaml#L144C27-L144C42)

### CODE REVIEW

**Review by AnouckColson — APPROVED** [review]

File: search/index_external_sites.sh [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 cleanup(){
     echo "* CLEANING UP OLD DOCS INDEX"
-    rm -f output/docs.json
+    rm -f output/platform_index.json
+    rm -f output/friday_index.json
+
```
Comment (Mupsi) [inline search/index_external_sites.sh:7]: Nitpicking: unnecessary empty line

**Review by Mupsi — APPROVED** [review]
Thanks for the video explanation, makes the process MUCH easier. Nothing shocked me in the changes I saw, but of course with a thousand files there's always the risk of missing something. Congrats on this first white label docs!

File: sites/friday/build_docs.sh [hunk 2]
```diff
@@ -39,7 +39,7 @@ else
 fi
 
 echo "Moving the hugo binary from cache into the application."
-cp "${PLATFORM_CACHE_DIR}/hugo_${HUGOVERSION}/hugo" "${PLATFORM_APP_DIR}"
+cp "${PLATFORM_CACHE_DIR}/hugo_${HUGOVERSION}/hugo" "${PLATFORM_APP_DIR}/sites/friday"
```
Comment (gilzow) [inline sites/friday/build_docs.sh:42]: I wonder if there's a way we can move the hugo binary still into `${PLATFORM_APP_DIR}` but then pass it a parameter for each site? Or maybe we copy the binary into  `${PLATFORM_APP_DIR}` but then create symlinks from each site back to the root binary?

Comment (gilzow) [reply sites/friday/build_docs.sh:42]: seems wasteful to have 2 copies of the same binary

File: build_local.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
```
Comment (gilzow) [inline build_local.sh:2]: we might add a check+warning concerning the need for a `GITHUB_API_TOKEN` env var, so someone doesnt get all the way through the process and then hit that issue once the npm scripts run

**Review by gilzow — APPROVED** [review]
some things we might want co change/clean-up later, but otherwise 👍

**Pull request merged**